optimum/amd/brevitas/quantizer.py

```diff
@@ -0,0 +1,342 @@
+            quantize_weight_zero_point=quantization_config.quantize_zero_point,
+            input_bit_width=quantization_config.activations_bitwidth,
+            input_quant_type="sym" if quantization_config.activations_symmetric else "asym",
+            input_quant_format="int",
+            input_param_method=quantization_config.activations_param_method,
+            input_scale_precision=quantization_config.scale_precision,
+            input_scale_type="static" if quantization_config.is_static else "dynamic",
+            input_quant_granularity=quantization_config.activations_quant_granularity,
+            input_group_size=quantization_config.activations_group_size,
+            quantize_input_zero_point=quantization_config.quantize_zero_point,
+        )
+
+        # Perform a single inference pass to generate the correct state_dict. This is necessary as Brevitas has some magic where
+        # a first forward pass need to be called before quantizing a model:
+        # https://github.com/Xilinx/brevitas/blob/84f42259ec869eb151af4cb8a8b23ad925f493db/src/brevitas/core/scaling/standalone.py#L205-L217
+        with torch.no_grad():
+            if calibration_dataset is not None:
+                model(**calibration_dataset[0])
+            elif not isinstance(model, torch.fx.GraphModule):
+                model(input_ids=torch.tensor([[1]], dtype=torch.int64))
+            else:
+                # TODO: clean that later.
+                config = AutoConfig.from_pretrained(self.model_name_or_path)
+
+                normalized_config_class = NormalizedConfigManager.get_normalized_config_class(config.model_type)
+                normalized_config = normalized_config_class(config)
+
+                num_heads = normalized_config.num_attention_heads
+                head_dim = normalized_config.hidden_size // num_heads
+                num_layers = normalized_config.num_layers
+
+                sample = {
+                    "input_ids": torch.tensor([[1]], dtype=torch.int64),
+                    "attention_mask": torch.tensor([[1]], dtype=torch.int64),
+                }
+                sample["past_key_values"] = tuple(
+                    (
+                        torch.zeros(1, num_heads, 0, head_dim, device=sample["input_ids"].device),
+                        torch.zeros(1, num_heads, 0, head_dim, device=sample["input_ids"].device),
+                    )
+                    for _ in range(num_layers)
+                )
+
+        if quantization_config.apply_gptq:
+            logger.info("Applying gptq...")
+            apply_gptq(
+                model,
+                calibration_dataset,
+                act_order=quantization_config.gptq_act_oder,
+                group_of_parallel_layers=self.group_of_parallel_layers,
+            )
+            logger.info("GPTQ applied.")
+
+        if quantization_config.activations_bitwidth is not None and quantization_config.is_static:
```

Comment:
Which is the case when `quantization_config.is_static is True` but bitwitdth is None

Reply:
It got lost in the changes but it is possible to pass activation_bitwidth as None to have weight-only quantization. It should be restored.